test/lib/Controller.test.ts

```diff
@@ -13,132 +13,147 @@ describe('Controller.ts tests', () => {
-		const engineOptions = new Map<string, string>([
-			[CUSTOM_CONFIG.EslintConfig, "/some/path"],
-			[CUSTOM_CONFIG.PmdConfig, "/some/other/path"]
-		]);
-
-		const engines: RuleEngine[] = await Controller.getEnabledEngines(engineOptions);
-		const names: string[] = engines.map(e => e.constructor.name);
-
-		expect(engines.length).to.equal(5);
-		expect(names).to.contain('CustomEslintEngine');
-		expect(names).to.contain('CustomPmdEngine');
-		expect(names).to.contain('RetireJsEngine');
-		expect(names).to.contain('SfgeDfaEngine');
-		expect(names).to.contain('SfgePathlessEngine');
-	});
-
-	it('getFilteredEngines filters and includes disabled', async() => {
-		const engines: RuleEngine[] = await Controller.getFilteredEngines([ENGINE.ESLINT, ENGINE.ESLINT_LWC, ENGINE.PMD]);
-		const names: string[] = engines.map(e => e.getName());
-
-		expect(engines.length).to.equal(3);
-		expect(names).to.contain(ENGINE.ESLINT);
-		expect(names).to.contain(ENGINE.ESLINT_LWC);
-		expect(names).to.contain(ENGINE.PMD);
-	});
-
-	it('getFilteredEngines uses custom config information to choose the correct instance', async() => {
-		const engineOptionsWithPmdCustom = new Map<string, string>([
-			[CUSTOM_CONFIG.PmdConfig, '/some/path/to/config']
-		]);
-		const engines: RuleEngine[] = await Controller.getFilteredEngines([ENGINE.PMD], engineOptionsWithPmdCustom);
-		const names: string[] = engines.map(e => e.getName());
-
-		expect(engines.length).to.equal(1);
-		expect(names).to.contain(ENGINE.PMD_CUSTOM);
-	});
-
-	it('getEnabledEngines throws exception when no engines are found', async() => {
-		// Create a single mocked engine that is disabled
-		const mockedRuleEngine: RuleEngine = mock<RuleEngine>();
-		when(mockedRuleEngine.getName).thenReturn(() => 'fake-engine');
-		when(mockedRuleEngine.isEnabled).thenReturn(() => Promise.resolve(false));
-		const ruleEngine: RuleEngine = instance(mockedRuleEngine);
-
-		// Remove everything else from the container and register the mock engine
-		container.reset();
-		container.registerInstance(Services.RuleEngine, ruleEngine);
-
-		try {
-			await Controller.getEnabledEngines();
-			fail('getEnabledEngines should have thrown');
-		} catch (e) {
-			expect(e.message).to.equal('You must enable at least one engine. Your currently disabled engines are: fake-engine.');
-		}
-	});
-
-	it('getFilteredEngines throws exception when no engines are found', async() => {
-		try {
-			await Controller.getFilteredEngines(['invalid-engine']);
-			fail('getFilteredEngines should have thrown');
-		} catch (e) {
-			expect(e.message).to.equal(`The filter doesn't match any engines. Filter 'invalid-engine'. Engines: cpd, eslint, eslint-lwc, eslint-typescript, pmd, retire-js, sfge.`);
-		}
+	describe('#getFilteredEngines()', () => {
+		it('If no filtering is provided, only requested-by-default engines are returned', async () => {
```

Comment:
This is the only new test in the entire file. It's making sure that the engines we expect to be requested-by-default actually are, and the rest aren't.